### DISCUSSION TITLE
Searching keyword in string using Client Filter

### DISCUSSION THREAD

**Discussion by mddanish00** [tinacms/tinacms#4312] [Help] [answered]
I have a Post collection with a title and body field. I want to create a search widget on my website that will search post titles that include the search keywords, so, I am looking at filters but... looking at this [documentation](https://tina.io/docs/data-fetching/overview/#filtering). 

I don't see any filtering operators that will help me with this. What I want like JavaScript String.includes...

Is this actually possible in TinaCMS? Maybe I misread? Or, it simply not possible right now?

**Comment by kldavis4** [comment, marked as answer]
That's correct. Substring search is not currently possible. Probably something like algolia is what you want for a search widget at the moment.

**Reply by mddanish00** [reply]
Well, that's unfortunate. I hope that will be implemented in the future. I look up Algolia but it feels too overpowered for a simple dev blog. I guess I could do without the search widget for the time being. 

Thanks for the answer!

**Reply by KitsonBroadhurst** [reply]
Agreed, I found the docs a little confusing and didn't realise that adding the search index only allowed admins to search.

**Comment by kldavis4** [comment]
fyi @jamespohalloran this is the second request I've seen recently where public search tokens would be useful. Here's the other: https://discord.com/channels/835168149439643678/1169558767084507197
